pkg/controller/controller_test.go

```diff
@@ -125,7 +124,7 @@ func createMockServer(t *testing.T, tmpdir string) (*gomock.Controller,
 }
 
 func tempDir(t *testing.T) string {
-	dir, err := ioutil.TempDir("", "external-provisioner-test-")
+	dir, err := os.MkdirTemp("", "external-provisioner-test-")
```

Comment:
Perhaps `t.TempDir()` and remove `defer` somewhere below?

Reply:
@testwill can you please fix this? The pr is OK otherwise.

Reply:
I am terribly sorry, I was not aware that go does not support unix domain socket paths longer than 107 characters: https://github.com/golang/go/issues/6895.

`t.TempDir` creates paths with the test name in them and they're quite long. You can see the test failures in CI that I was waiting for you to fix.

Can you please go back to `os.MkdirTemp`?